components/tyk-gateway/templates/deployment-gw-repset.yaml

```diff
@@ -338,9 +338,18 @@ spec:
             value: "{{ .Values.gateway.opentelemetry.exporter }}"
           - name: TYK_GW_OPENTELEMETRY_ENDPOINT
             value: "{{ .Values.gateway.opentelemetry.endpoint}}"
+            {{- if and .Values.gateway.opentelemetry.headers .Values.gateway.opentelemetry.headerSecret.name }}
+              {{- fail "You can't use '.Values.gateway.opentelemetry.headers' and '.Values.gateway.opentelemetry.headerSecret' at the same time!" }}
+            {{- end }}
             {{- if .Values.gateway.opentelemetry.headers }}
           - name: TYK_GW_OPENTELEMETRY_HEADERS
             value: "{{ include "otel-headers" . }}"
+            {{- else if and .Values.gateway.opentelemetry.headerSecret.name .Values.gateway.opentelemetry.headerSecret.key }}
+          - name: TYK_GW_OPENTELEMETRY_HEADERS
```

Comment:
[TYK_GW_OPENTELEMETRY_HEADERS](https://tyk.io/docs/tyk-oss-gateway/configuration/#opentelemetryheaders) environment variable needs to be declared as a comma-separated list of key-value pairs, where each pair is separated by a colon. For instance: 
```
TYK_GW_OPENTELEMETRY_HEADERS: "header_key1:header_value1,header_key2:header_value2"
```

So, we need to construct the value of the environment variable in this key-value pair format.